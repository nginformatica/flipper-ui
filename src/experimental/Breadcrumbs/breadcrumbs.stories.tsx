import React from 'react'
import BreadcrumbsComponent from '.'
import { Meta, StoryFn } from '@storybook/react'

export default {
    title: 'experimental/Breadcrumbs',
    component: BreadcrumbsComponent
} as Meta<typeof BreadcrumbsComponent>

const Template: StoryFn<typeof BreadcrumbsComponent> = args => {
    return <BreadcrumbsComponent {...args} />
}

export const Default = Template.bind({})
Default.args = {
    pathname: '/foo/bar',
    homeId: '-',
    pathMapping: {
        '-': 'Home',
        foo: 'Foo',
        bar: 'Bar'
    }
}
