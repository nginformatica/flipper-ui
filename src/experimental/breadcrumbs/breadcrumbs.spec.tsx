import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Breadcrumbs from '.'

describe('Breadcrumbs', () => {
    it('should render', () => {
        render(
            <Breadcrumbs
                pathname={'/foo/bar'}
                homeId='-'
                pathMapping={{
                    '-': 'Home',
                    foo: 'Foo',
                    bar: 'Bar'
                }}
            />
        )

        const breadcrumbs = screen.getByText('Home')
        expect(breadcrumbs).toBeDefined()
    })

    it('should render without pathname', () => {
        render(
            <Breadcrumbs
                homeId='-'
                pathMapping={{
                    '-': 'Home',
                    foo: 'Foo',
                    bar: 'Bar'
                }}
            />
        )

        const breadcrumbs = screen.getByText('Home')
        expect(breadcrumbs).toBeDefined()
    })
})
