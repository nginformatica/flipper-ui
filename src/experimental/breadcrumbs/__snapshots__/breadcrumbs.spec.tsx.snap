// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should match snapshot 1`] = `
<div>
  <nav
    class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    <ol
      class="MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="sc-EFWon euHyUG"
          href="/"
        >
          <p
            class="MuiTypography-root sc-fYlSAf hZrMuA MuiTypography-body2 MuiTypography-colorInherit"
          >
            Home
          </p>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="sc-EFWon euHyUG"
          href="/foo"
        >
          <p
            class="MuiTypography-root sc-fYlSAf hZrMuA MuiTypography-body2 MuiTypography-colorInherit"
          >
            Foo
          </p>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <p
          class="MuiTypography-root sc-fYlSAf hZrMuA MuiTypography-body2 MuiTypography-colorInherit"
        >
          Bar
        </p>
      </li>
    </ol>
  </nav>
</div>
`;
